### PR TITLE
ChainSync client: remove redundant intersection check

### PR DESCRIPTION
**Pull request by amesgen**
This step has caused confusions in the past.

See the commit description of the single commit.

### CODE REVIEW

File: ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/ChainSync/Client.hs [hunk 1]
```diff
@@ -1108,13 +1107,7 @@ knownIntersectionStateTop cfgEnv dynEnv intEnv =
         recordHeaderArrival
       } = headerInFutureCheck
 
-    -- Request the next message (roll forward or backward), unless our chain
```
Comment (nfrisby) [inline ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/ChainSync/Client.hs:1111]: What do you think about adding a comment on the MsgRollForward handler explaining why it's sufficient for it to be the only place that checks for the intersection?

Comment (amesgen) [reply ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/ChainSync/Client.hs:1111]: Added a comment to `rollForward` :+1:

**Comment by facundominguez** [comment]
Not sure if it is relevant, but in #1033 (3164620) I found a trace were the removed code sets the candidate fragment to `G` from `G | 1-0`. [This](https://github.com/IntersectMBO/ouroboros-consensus/actions/runs/8853931627/job/24315856092?pr=1033) this is the failure. And I'm attaching the [trace](https://github.com/IntersectMBO/ouroboros-consensus/files/15136755/out.txt) where you can see that at tick 44, adversary 2 gets its candidate fragment set to G.

I'm not sure it is relevant because it could be a CSJ artifact.

**Comment by amesgen** [comment]
> the removed code sets the candidate fragment to `G` from `G | 1-0`.

I don't think this can be the case; the code that is removed in this PR never changes the tip of the candidate; it only ever causes us to disconnect or to change the *anchor* of the candidate fragment.

**Comment by facundominguez** [comment]
> I found a trace were the removed code sets the candidate fragment to `G` from `G | 1-0`

Today, with additional tracing, looks like this is not the case. In all likelihood, it is a false alarm.

**Review by nfrisby — APPROVED** [review]
Thanks for the simplification!

**Pull request merged**